src/tests/ftest/cart/tests_common.h

```diff
@@ -498,4 +493,75 @@ tc_srv_start_basic(char *srv_group_name, crt_context_t *crt_ctx,
 	D_ASSERTF(rc == 0, "crt_group_size() failed; rc=%d\n", rc);
 }
 
+struct tc_log_msg_cb_resp {
+	sem_t	sem;
+};
+
+static void
+tc_log_msg_cb(const struct crt_cb_info *info)
+{
+	struct tc_log_msg_cb_resp	*resp;
+
+	if (info->cci_rc != 0) {
+		D_WARN("Add Log message CB failed\n");
+		D_ASSERTF(info->cci_rc == 0,
+			  "Send Log RPC did not respond\n");
+	}
+	resp = (struct tc_log_msg_cb_resp *)info->cci_arg;
+	sem_post(&resp->sem);
+}
+
+int
+tc_log_msg(crt_context_t ctx, crt_group_t *grp, d_rank_t rank,
+	   char *msg) {
+	int32_t				 rc = 0;
+	struct crt_ctl_log_add_msg_in	*send_args;
+	crt_rpc_t			*rpc_req = NULL;
+	crt_endpoint_t			 ep;
+	crt_opcode_t			 opcode = CRT_OPC_CTL_LOG_ADD_MSG;
+	struct tc_log_msg_cb_resp	 resp;
+
+	/* Initialize response structure */
+	rc = sem_init(&resp.sem, 0, 0);
+	D_ASSERTF(rc == 0, "sem_init() failed\n");
+
+	/* Fill in the endpoint info */
+	ep.ep_grp = grp;
+	ep.ep_rank = rank;
+	ep.ep_tag = 0;
+
+	rc = crt_req_create(ctx, &ep, opcode, &rpc_req);
+	if (rc != 0) {
+		D_ERROR("crt_req_create() failed. rc %d.\n", rc);
+		D_GOTO(exit, rc);
+	}
+
+	crt_req_addref(rpc_req);
+	send_args =  crt_req_get(rpc_req);
+	send_args->log_msg = msg;
+
+	/* send the request */
+	rc = crt_req_send(rpc_req, tc_log_msg_cb, &resp);
+	if (rc < 0) {
+		D_WARN("rpc failed, message: \"%s \"not sent\n", msg);
+		goto cleanup;
+	}
+
+	/* Wait for response */
+	rc = tc_sem_timedwait(&resp.sem, 30, __LINE__);
+	if (rc < 0) {
+		D_WARN("Messaage logged timed out: %s\n", msg);
+		crt_req_abort(rpc_req);
+		goto cleanup;
+	} else {
+		D_INFO("Message logged to rank %d:  %s\n", rank, msg);
+	}
```

Comment:
i dont think we want to print this part. if successful the message will be already emitted in the server log, we dont want to duplicate it in the client log as well. if any test client wants to do such duplication they can do it outside of the call of this helper function, but we shouldnt do it automatically

Reply:
Removed.